docs/book/src/plugins/creating-plugins.md

```diff
@@ -64,3 +64,46 @@ Feel free to check the implementation under:
 
 - deploy-image: <https://github.com/kubernetes-sigs/kubebuilder/tree/v3.7.0/pkg/plugins/golang/deploy-image/v1alpha1>
 - grafana: <https://github.com/kubernetes-sigs/kubebuilder/tree/v3.7.0/pkg/plugins/optional/grafana/v1alpha>
+
+## Plugin scaffolding
+
+Your plugin may have the use case to add code on top of what is scaffolded by default with `create api`.
+This is common as you may expect your plugin to:
```

Comment:
WDYT about we have here something like "Helpers in the API to do the scaffolds" and then we describe the options with something like:



### Bollerplates

You can check in the code example that the plugins will have boilerplates to generate
the files with code. Therefore, you can see that Kubebuider API provides some utils which
can help you to achieve your goals.

// TODO: Add an example of template

### You can overwrite a file

You might want for example to overwrite a scaffold done by using the option:

```
 f.IfExistsAction = machinery.OverwriteFile 
```

Example:

Let's imagine that you would like to have a helper plugin that would be called in a chain with go/v4-alpha to add customizations on top/ Therefore after we generate the code calling the subcommand to init from go/v4 alpha we would like to overwrite the Makefile to change this scaffold via our plugin. In this way, we would implement the Bollerplate for our Makefile and then use this option to ensure that it would be overwritten.